zpay32/invoice.go

```diff
@@ -294,8 +294,8 @@ func Decode(invoice string, net *chaincfg.Params) (*Invoice, error) {
 	}
 	decodedInvoice.Net = net
 
-	// If the HRP is longer than "ln" + the segwit prefix, parse the
-	// payment amount.
+	// Optionally, if there's anything left of the HRP after ln + the segwit
+	// prefix, it parses the payment amount.
```

Comment:
s/parses/encodes

Reply:
@halseth isn't "parses" the appropriate term here? It's not encoding the amount, it's reading it from a part of a string. "Decodes" might work too but I figure the HRP doesn't technically get decoded.

Reply:
Ah, I see why we disagree 😛  In this context, I feel "it" references the remainder of the HRP.

Reply:
Ahh yeah. How about an (hopefully) unambiguous "Optionally, if there's anything left of the HRP after ln, the remainder is decoded to an amount."

In reconsidering this, "decode" actually makes more sense than "parse" since the amount is encoded with an SI unit and it's not as simple as just reading a plain number, it also matches the `decodeAmount` function a few lines below.

Reply:
What about "Optionally, if there's anything left of the HRP after ln + the segwit prefix, we try to decode this as the payment amount"? :)